### PR TITLE
Account\Pcopy Command

### LINKED ISSUE #22921: pcopy command
**Description:**

pcopy command will be the fast way for pdump + pload

**Current behaviour:**

When we have to test bugs we must levelup characters, equip them, learn talents and spells or pdump a character and load to one new account

**Expected behaviour:**

with pcopy it will allow use pcopy "source character" "target account" we will have a copy of one existing character.

**Steps to reproduce the problem:**

1. go test some bug

**Branch(es):**

CHANGEME both

**TC rev. hash/commit:** 

N/R d12f0841f198

**Operating system:** N/R

### PR DESCRIPTION
**Pull request by Lorenc95**
**Changes proposed:**

-  Adds a command to copy a character
-  
-  

**Target branch(es):** 3.3.5/master

- [√] 3.3.5
- [ ] master

**Issues addressed:** Closes #22921 

**Tests performed:** None

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] add the actual copying of the character to the new account
- [ ] Leave the actual character and account owner unharmed/untouched/unaltered.
- [ ] Use the same GUID? if not, Copy all character info. 
- [ ] Force the rename on the character using a DB query and update the value of forcerename

This is just a quick edit, will update, was to get the general idea of what i had in mind going!!

### CODE REVIEW

**Comment by mdX7** [comment]
`Use the same GUID? if not, Copy all character info.`
Using the same GUID of the char being copied shouldn't be possible since it already exists. Or I did not get what you mean :D

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -9563,7 +9563,7 @@ void Unit::UpdateCharmAI()
         UnitAI* newAI = nullptr;
         if (GetTypeId() == TYPEID_PLAYER)
         {
-            if (Unit* charmer = GetCharmer())
+            if (Unit* charmer = GetCharmer() || GetCharmer() == TYPEID_PLAYER)
```
Comment (Aokromes) [inline src/server/game/Entities/Unit/Unit.cpp]: i suppose not intended change

Comment (illfated) [reply src/server/game/Entities/Unit/Unit.cpp]: Anyway, that change is the reason why the build fails.

Comment (IllInuz) [reply src/server/game/Entities/Unit/Unit.cpp]: Non intented change, will revert.

File: src/server/scripts/Commands/cs_misc.cpp [hunk 2]
```diff
@@ -116,6 +116,7 @@ class misc_commandscript : public CommandScript
             { "unstuck",          rbac::RBAC_PERM_COMMAND_UNSTUCK,           true, &HandleUnstuckCommand,          "" },
             { "wchange",          rbac::RBAC_PERM_COMMAND_WCHANGE,          false, &HandleChangeWeather,           "" },
             { "mailbox",          rbac::RBAC_PERM_COMMAND_MAILBOX,          false, &HandleMailBoxCommand,          "" },
+            { "pcopy",            rbac::RBAC_PERM_COMMAND_PCOPY,            false, &HandlPCopyCommand,          "" },
```
Comment (kuskmen) [inline src/server/scripts/Commands/cs_misc.cpp]: Typo here should be &HandlePCopyCommand?

**Comment by kuskmen** [comment]
What happened with this PR why it is closed?

**Comment by illfated** [comment]
Only @IllInuz knows what actually happened. Maybe something went wrong during his attempt to fix the mistake above? Maybe he was unable to open a new PR or did not have enough time? I don't know.

**Pull request closed without merging**